pkg/virt-launcher/virtwrap/device/hostdevice/generic/hostdev.go

```diff
@@ -34,9 +35,95 @@ const (
 	DefaultDisplayOff                 = false
 )
 
-func CreateHostDevices(vmiHostDevices []v1.HostDevice) ([]api.HostDevice, error) {
-	return CreateHostDevicesFromPools(vmiHostDevices,
+func CreateHostDevices(vmi *v1.VirtualMachineInstance) ([]api.HostDevice, error) {
```

Comment:
Just to be extra safe I would do

```
if !hasHostDevicesWithDRA() && !hasHostDevicesWithDRA{
   return  CreateHostDevicesFromPools(vmiHostDevices,
		NewPCIAddressPool(vmiHostDevices), NewMDEVAddressPool(vmiHostDevices), NewUSBAddressPool(vmiHostDevices))
}
```

Reply:
updated by moving the DRA code into a new package.